src/leaderboard/leaderboard.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@nestjs/common';
+import { ServiceHelper } from 'src/common/helpers/service.helper';
+import { LeaderboardRepository } from './repository/leaderboard.repository';
+import { LeaderboardDto } from './dto/leaderboard.dto';
+import { LeaderboardEntity as Leaderboard } from './entities/leaderboard.entity';
+import { PlayerLeaderboardRepository } from 'src/player-leaderboard/repository/player-leaderboard.repository';
+import { SortedResult } from './dto/sorted-result.dto';
+import { SortingOrders } from './entities/sorting.enum';
+import { PlayerLeaderboardEntity as PlayerLeaderboard } from 'src/player-leaderboard/entity/player-leaderboard.entity';
+
+@Injectable()
+export class LeaderboardService {
+  constructor(
+    private readonly serviceHelper: ServiceHelper,
+    private readonly leaderboardRepository: LeaderboardRepository,
+    private readonly playerLeaderboardRepository: PlayerLeaderboardRepository,
+  ) {}
+
+  async createLeaderboard(id: string | undefined, data: LeaderboardDto): Promise<Leaderboard> {
+    const fields: { [l: string]: any } = { ...data };
+    const newLeaderboard: Leaderboard = await this.serviceHelper.getUpsertData(id, fields, this.leaderboardRepository);
+    return this.leaderboardRepository.save(newLeaderboard);
+  }
+
+  async sortLeaderboard(
+    leaderboardId: string,
+    leaderboardName: string,
+    metrics: string[],
+    sortingOrder: SortingOrders[],
```

Comment:
Get `Leaderboard` by `id` from the database instead of passing `metrics` and `sortingOrder`.